web/packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+import { ButtonBorder, Text, Box, Menu, MenuItem } from 'design';
+import { CarrotDown } from 'design/Icon';
+
+import cfg from 'teleport/config';
+import { ParticipantMode } from 'teleport/services/session';
+
+export const SessionJoinBtn = ({
+  sid,
+  clusterId,
+  participantModes,
+}: {
+  sid: string;
+  clusterId: string;
+  participantModes: ParticipantMode[];
+}) => {
+  // Sorts the list of participantModes so that they are consistently shown in the order of "observer" -> "moderator" -> "peer"
+  const modes = {
+    observer: 1,
+    moderator: 2,
+    peer: 3,
+  };
+  const sortedParticipantModes = participantModes.sort(
+    (a, b) => modes[a] - modes[b]
+  );
+
+  return (
+    <JoinMenu>
+      {sortedParticipantModes.map(participantMode => (
+        <MenuItem
```

Comment:
At one point we talked about a tooltip or help text that would explain the various modes. Feel free to do that in a separate PR so as not to hold this up, but I still think it's worth doing.